Assignment 2/src/main/java/com/nuclei/assignment/service/inputparser/InputParserImpl.java

```diff
@@ -0,0 +1,101 @@
+package com.nuclei.assignment.service.inputparser;
+
+import com.nuclei.assignment.constants.ExceptionsConstantsUtils;
+import com.nuclei.assignment.enums.Courses;
+import com.nuclei.assignment.enums.SortingOrder;
+import com.nuclei.assignment.exception.CustomException;
+import com.nuclei.assignment.service.inputvalidation.InputValidation;
+import com.nuclei.assignment.service.inputvalidation.InputValidationImpl;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Set;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * User Parse Implementation containing method to parse different properties.
+ */
+public class InputParserImpl implements InputParser {
+  
+  /**
+   * The Logger.
+   */
+  private final Log logger = LogFactory.getLog(InputParserImpl.class);
+  
+  /**
+   * The Input validation.
+   */
+  private final InputValidation inputValidation;
+  
+  /**
+   * Instantiates a new Input parser.
+   */
+  public InputParserImpl() {
+    inputValidation = new InputValidationImpl();
+  }
+  
+  @Override
+  public String parseName(final String name) throws CustomException {
+    inputValidation.checkDataIsNull(name, "Name");
```

Comment:
catch the exception and modify the message to be more meaningful, instead of passing parameter here.

Reply:
done